server/src/main/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorService.java

```diff
@@ -207,5 +217,42 @@ public SimpleHealthIndicatorDetails getDetails() {
                 )
             );
         }
+
+        public List<HealthIndicatorImpact> getImpacts() {
+            final List<HealthIndicatorImpact> impacts = new ArrayList<>();
+            if (primaries.indicesWithUnavailableShards.isEmpty() == false) {
+                String impactDescription = String.format(
+                    Locale.ROOT,
+                    "Cannot add data to %d %s [%s]. Searches might return incomplete results.",
+                    primaries.indicesWithUnavailableShards.size(),
+                    primaries.indicesWithUnavailableShards.size() == 1 ? "index" : "indices",
+                    getTruncatedIndicesString(primaries.indicesWithUnavailableShards)
+                );
+                impacts.add(new HealthIndicatorImpact(1, impactDescription));
+            }
+            Set<String> indicesWithUnavailableReplicasOnly = new HashSet<>(replicas.indicesWithUnavailableShards);
+            indicesWithUnavailableReplicasOnly.removeAll(primaries.indicesWithUnavailableShards);
```

Comment:
Can you please shortly document **why** we do this?

Reply:
Done